src/monocypher.h

```diff
@@ -57,7 +57,9 @@
 #include <stddef.h>
 #include <stdint.h>
 
-#ifdef __cplusplus
+#ifdef MONOCYPHER_CPP_NAMESPACE
+namespace MONOCYPHER_CPP_NAMESPACE {
+#elif defined(__cplusplus)
```

Comment:
That's a tough call. On the one hand, this flag looks harmless enough, and covers a real use case. On the other hand it's yet another flag, and I'd like to keep them to a minimum. Also, the fact that any competent C/C++ dev can just replace the `extern "C"` by a namespace, makes the flag a bit harder to justify. I mean it's not like you'd have to touch any cryptographic code. On the other hand, people are so scared of touching any crypto with a 10 foot pole that it might be worth signaling that it's a valid use case. @fscoto, thoughts?

Also, if we're to add that flag, the deprecated files will need some love too. :-)